trp_project_scrum/tests/test_trp_project_scrum.py

```diff
@@ -0,0 +1,168 @@
+        })
+        self.task5 = self.env['project.task'].create({
+           'name' : 'task_noscrum',
+           'project_id' : self.not_scrum_project.id
+        })
+
+
+    def test_trp_project_scrum_permissions(self):
+        all_projects = [
+             self.scrumproject_alpha,
+             self.scrumproject_beta,
+             self.scrumproject_gamma,
+             self.not_scrum_project
+        ]
+        all_tasks = [
+            self.task1,
+            self.task2,
+            self.task3,
+            self.task4,
+            self.task5
+        ]
+        #check if other groups are assigned
+        self.assertEqual(self.collaborator_follower.has_group(
+            'trp_external_user.group_external_user'
+        ), True,
+        'Group External User was not triggered after assigning'
+        ' collaborator group'
+        )
+
+        self.assertEqual(self.collaborator_follower.has_group(
+            'trp_project_scrum.collaborators_followers')
+        , True,
+        'Collaborator follower group was not triggered after assigning'
+        )
+        self.scrumproject_alpha.write({
+            'message_follower_ids' : [
+                (6, 0, [self.collaborator_follower.partner_id.id])
+            ]
+        })
+        # verify it cannot see any element of other projects
+        projects = self.env['project.project'].sudo(
+           self.collaborator_follower.id
+        ).search([('id', 'in', [x.id for x in all_projects])])
+        # should see only 1 project
+        self.assertEqual(
+            len(projects), 1,
+            'a collaborator follower sees too many projects'
+        )
+        # verify admin can see all still all
+        projects2 = self.env['project.project'].sudo().search(
+            [('id', 'in', [x.id for x in all_projects])]
+        )
+        # should see all  projects we made
+        self.assertEqual(
+            len(projects2), 4, 'admin does not see all prgs'
+        )
+        # verify user cannot modify projects or scrum elements he can see
+        #collaborator can see no tasks:
+        tasks = self.env['project.task'].sudo(
+               self.collaborator_follower.id
+            ).search([('id', 'in', [x.id for x in all_tasks])])
+
+        # should see only the task of the project
+        self.assertEqual(
+            len(tasks), 2,
+            'Collaborator can see some tasks when should see none'
+        )
+
+        # verify that he can modify tasks of his project
+        self.assertEqual(
+            self.task2.sudo(
+                self.collaborator_follower.id
+            ).with_context(mail_notrack=True).write({'name': 'this is ok'}),
+            True,
+            'Collaborator could not write on a task where has permissions'
+        )
+        self.assertRaises(
+            AccessError,
+            self.task5.sudo(
+                self.collaborator_follower.id
+            ).with_context(mail_notrack=True).write, {'name': 'this is not ok'}
```

Comment:
this works, but is a bit clumsy. The context manager version is a lot simpler to write and read i think:

```python
with self.assertRaises(AccessError):
    self.task5.sudo(self.collaborator_follower.id)\
        .with_context(mail_notrack=True)\
        .write({'name': 'this is not ok'})
```

(nice writeup about context managers here: https://jeffknupp.com/blog/2016/03/07/python-with-context-managers/)